### PR TITLE
Adjusting showYahooPrompt for New Experiment

### PR DESCRIPTION
**Pull request by jedtan**

### CODE REVIEW

File: graphql/database/users/__tests__/getShouldShowYahooPrompt.test.js [hunk 1]
```diff
@@ -50,17 +68,47 @@ describe('shouldShowYahooPrompt tests', () => {
...
-        variation: true,
-        yahooPaidSearchRewardOptIn: false,
-      })
-    )
+    getUserFeature
+      .mockResolvedValueOnce(
+        new Feature({
+          featureName: YAHOO_SEARCH_EXISTING_USERS,
+          variation: true,
+          yahooPaidSearchRewardOptIn: false,
```
Comment (kmjennison) [inline graphql/database/users/__tests__/getShouldShowYahooPrompt.test.js:80]: This shouldn't have the `yahooPaidSearchRewardOptIn` property, right?

File: graphql/database/users/__tests__/getShouldShowYahooPrompt.test.js [hunk 2]
```diff
@@ -50,17 +68,47 @@ describe('shouldShowYahooPrompt tests', () => {
...
+  it('returns true if users feature is true for new user test', async () => {
+    expect.assertions(1)
+    const getShouldShowYahooPrompt = require('../getShouldShowYahooPrompt')
+      .default
+    getUserFeature
+      .mockResolvedValueOnce(
+        new Feature({
+          featureName: YAHOO_SEARCH_EXISTING_USERS,
+          variation: findLastIndex,
+          yahooPaidSearchRewardOptIn: false,
```
Comment (kmjennison) [inline graphql/database/users/__tests__/getShouldShowYahooPrompt.test.js:103]: Same as above

File: graphql/database/users/__tests__/getShouldShowYahooPrompt.test.js [hunk 3]
```diff
@@ -50,17 +68,47 @@ describe('shouldShowYahooPrompt tests', () => {
...
+
+  it('returns true if users feature is true for new user test', async () => {
+    expect.assertions(1)
+    const getShouldShowYahooPrompt = require('../getShouldShowYahooPrompt')
+      .default
+    getUserFeature
+      .mockResolvedValueOnce(
+        new Feature({
+          featureName: YAHOO_SEARCH_EXISTING_USERS,
+          variation: findLastIndex,
```
Comment (kmjennison) [inline graphql/database/users/__tests__/getShouldShowYahooPrompt.test.js:102]: What's the effect of `findLastIndex` here?

**Review by kmjennison — CHANGES_REQUESTED** [review]
Looks good, just a few questions about tests.

File: graphql/database/users/__tests__/getShouldShowYahooPrompt.test.js (hunk 3 above)
Comment (jedtan) [reply graphql/database/users/__tests__/getShouldShowYahooPrompt.test.js:102]: No idea what was going on here. Will update real quick.

File: graphql/database/users/__tests__/getShouldShowYahooPrompt.test.js [hunk 4]
```diff
@@ -14,12 +18,19 @@ describe('shouldShowYahooPrompt tests', () => {
...
+          featureName: YAHOO_SEARCH_EXISTING_USERS,
+          variation: false,
+        })
+      )
+      .mockResolvedValueOnce(
+        new Feature({
+          featureName: YAHOO_SEARCH_NEW_USERS,
+          variation: 'Google',
+        })
+      )
```
Comment (kmjennison) [inline graphql/database/users/__tests__/getShouldShowYahooPrompt.test.js:33]: This test says it's testing when the user is **in** the experiment, so should one of these features be the experiment variation?

Comment (kmjennison) [reply graphql/database/users/__tests__/getShouldShowYahooPrompt.test.js:33]: Also, is this test intending to do the same thing as the one below?

**Review by kmjennison — CHANGES_REQUESTED** [review]

**Pull request merged**